### PR TITLE
Support JDK client in perf tests

### PR DESCRIPTION
**Pull request by lmolkova**
Add support for JDK client in perf tests  - the key part is not to fail when it's passed in `--http-client`.

Insecure configuration is not possible since `net.java.http.HttpClient` is not available on Java 8.
It's still possible to configure it in other projects that target higher Java versions which can configure their clients however they want to.

### CODE REVIEW

File: common/perf-test-core/pom.xml [hunk 1]
```diff
@@ -81,5 +81,10 @@
       <artifactId>azure-core-http-okhttp</artifactId>
       <version>1.11.17</version> <!-- {x-version-update;com.azure:azure-core-http-okhttp;dependency} -->
     </dependency>
+    <dependency>
```
Comment (alzimmermsft) [inline common/perf-test-core/pom.xml:84]: Two questions:

- Can we also include Vert.x as well? We need to start getting performance numbers for it as well.
- Do we create the uber JAR correctly where it includes all the HttpClientProviders in `META-INF`?

Both can be punted to another PR if needed.

File: common/perf-test-core/src/main/java/com/azure/perf/test/core/PerfStressOptions.java [hunk 2]
```diff
@@ -144,6 +144,6 @@ public List<String> split(String value) {
     }
 
     public enum HttpClientType {
-        NETTY, OKHTTP
+        NETTY, OKHTTP, JDK
```
Comment (alzimmermsft) [inline common/perf-test-core/src/main/java/com/azure/perf/test/core/PerfStressOptions.java:147]: _Can be done in this PR or pushed to the future._

We should change this from being a magical string to being the full class name of the HttpClientProvider used to create the HttpClient instance. Or, at least give this enum a field for that value (`NETTY("com.azure.core.http.netty.NettyAsyncHttpClientProvider")`).

That way we can use the logic in our service provider loader to get the explicit instance we want instead of needing to use the `switch` in `ApiPerfTestBase` and should allow for us to make the code more generic and not need updating every time we add a new HttpClient.

**Review by alzimmermsft — APPROVED** [review]

File: common/perf-test-core/pom.xml (hunk 1 above)
Comment (lmolkova) [reply common/perf-test-core/pom.xml:85]: 1. Sure, added vertx.
2. It does not need SPI to create clients since it knows the type and I'd rather keep it this way to simplify things, especially for perf tests which use this jar

File: common/perf-test-core/src/main/java/com/azure/perf/test/core/PerfStressOptions.java (hunk 2 above)
Comment (lmolkova) [reply common/perf-test-core/src/main/java/com/azure/perf/test/core/PerfStressOptions.java]: Been thinking about it and in manual cmd line it's easier to write `--http-client jdk`, but what do you think about the new impl:
- known clients have friendly names (`netty`, `okhttp`, `vertx`, `jdk`). 
- we can also (in most cases) create instances that trust all certs
- we can create them without SPI (and don't need shading plugins, etc)
-  plus any full priovider name is supported if resolved with SPI

Tested it on core perf tests, works nicely:

`java -jar .\target\azure-core-perf-1.0.0-beta.1-jar-with-dependencies.jar binarydatareceive -w 0 -d 1 -p 1 --http-client com.azure.core.http.okhttp.OkHttpAsyncClientProvider  ...`

`java -jar .\target\azure-core-perf-1.0.0-beta.1-jar-with-dependencies.jar binarydatareceive -w 0 -d 1 -p 1 --http-client okhttp  ...`

**Review by alzimmermsft — APPROVED** [review]

**Review by g2vinay — APPROVED** [review]
